### PR TITLE
feat(UI) add topology to agent page

### PR DESCRIPTION
**Pull request by shazlehu**
### Proposed Change
Adds a configuration topology graph to the Agent page.

Resolves observIQ/Iris#1266 and observIQ/Iris#1297

##### Checklist
- [ ] Changes are tested
- [ ] CI has passed

### CODE REVIEW

**Comment by shazlehu** [comment]
TODO: display measurements for only the selected agent on the configuration graph.

**Comment by shazlehu** [comment]
Do we like how this is laid out?

image

Current layout:

image

**Review by dsvanlani — COMMENTED** [review]
This Editing view is a little weird:

image

- The drop down is supposed to be in the bottom box, which should grow to fill the page (I think)
- The top should still read "Configuration - <current_config_name>"
- I dont think we should put the error in the bottom box when editing:
image

**Review by dsvanlani — COMMENTED** [review]
I think this is supposed to be the dark blue color: 
image

**Review by dsvanlani — COMMENTED** [review]
Spacing is a little off between the switcher, error, and config 

image

Should be:

image

**Review by dsvanlani — CHANGES_REQUESTED** [review]
Final requests to change:

center details in box
align table rows with no value
remove yaml editor in manageconfig form
Dont display error while editing
Margin below Error Message is too large
Get rid of borderless card container, use box
Remove second box when editing config

**Comment by shazlehu** [comment]
Implemented all of Dave's changes.

File: ui/src/components/CardContainer/index.tsx [hunk 1]
```diff
@@ -1,11 +1,28 @@
...
+    </Paper>
+  );
+};
+
+export const BorderlessCardContainer: React.FC = ({ children }) => {
+  return (
+    <Paper
+      classes={{ root: styles.root }}
+      elevation={1}
+      sx={{ borderWidth: 0 }}
```
Comment (dsvanlani) [inline ui/src/components/CardContainer/index.tsx:24]: This background color should match the app body background.

```suggestion
      sx={{ borderWidth: 0, backgroundColor: "#fcfcfc" }}
```

**Review by dsvanlani — APPROVED** [review]
Just need to fix the background color on the new Card, looks clean, working great 👍

File: ui/src/components/Cards/InlineSourceCard.tsx [hunk 2]
```diff
@@ -73,11 +73,13 @@ export const InlineSourceCard: React.FC<{
   // For inline sources this is expected to be in the form source0, source1, etc
   id: string;
   disabled?: boolean;
-}> = ({ id, disabled }) => {
+  configuration: MinimumRequiredConfig;
+  refetchConfiguration: () => void;
+}> = ({ id, disabled, configuration, refetchConfiguration }) => {
   const sourceIndex = getSourceIndex(id);
-  const { configuration, refetchConfiguration } = useConfigurationPage();
+  //const { configuration, refetchConfiguration } = useConfigurationPage();
```
Comment (antonblock) [inline ui/src/components/Cards/InlineSourceCard.tsx:80]:
```suggestion
```

File: ui/src/components/PipelineGraph/ConfigurationFlow.tsx [hunk 3]
```diff
@@ -48,24 +54,33 @@ export const TARGET_OFFSET_MULTIPLIER = 250;
...
 export const ConfigurationFlow: React.FC<ConfigurationFlowProps> = ({
   period,
   selectedTelemetry,
+  configuration,
+  refetchConfiguration,
+  agent,
 }) => {
   const reactFlowInstance = useReactFlow();
-  const { configuration } = useConfigurationPage();
+  //const { configuration } = useConfigurationPage();
```
Comment (antonblock) [inline ui/src/components/PipelineGraph/ConfigurationFlow.tsx:70]:
```suggestion
```

File: ui/src/components/PipelineGraph/RawOrTopologyControl.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    _: React.MouseEvent<HTMLElement>,
+    newValue: "topology" | "raw",
+    ) => {
+      setTopologyOrRaw(newValue);
+  };
+  return (
+    <ToggleButtonGroup
+      sx={{
+        display: 'flex',
+        justifyContent: 'center',
```
Comment (antonblock) [inline ui/src/components/PipelineGraph/RawOrTopologyControl.tsx:22]:
```suggestion
        justifyContent: 'center',
```
There are a few lines in here with a bunch of trailing spaces

**Review by antonblock — APPROVED** [review]
Looks good, minus the small comments I added

**Pull request merged**